File: sdk/python/src/tinyverse/types/directory.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Any, Optional

from .identity import Identity, PaymentMethod


@dataclass
class AgentInterface:
    url: str
    binding: str
    version: str


@dataclass
class AgentPayment:
    network: str
    asset: str
    rate_type: str
    amount: str


@dataclass
class AgentDocs:
    swagger_json: Optional[str] = None
    swagger_md: Optional[str] = None
    skill_md: Optional[str] = None
    swagger_json_url: Optional[str] = None
    swagger_md_url: Optional[str] = None
    skill_md_url: Optional[str] = None


@dataclass
class AgentWebhook:
    event: str
    url: str
    secret_ref: Optional[str] = None
    description: Optional[str] = None
    metadata: Optional[dict[str, str]] = None


@dataclass
class AgentCard:
    agent_id: str
    name: str
    crypto_id: str
    created_at: str
    updated_at: str
    description: Optional[str] = None
    username: Optional[str] = None
    public_key: Optional[str] = None
    url: Optional[str] = None
    endpoint: Optional[str] = None
    supported_interfaces: Optional[list[AgentInterface]] = None
    skills: Optional[list[str]] = None
    capabilities: Optional[list[str]] = None
    tags: Optional[list[str]] = None
    payment_methods: Optional[list[PaymentMethod]] = None
    payment_requirements: Optional[AgentPayment] = None
    groups: Optional[list[str]] = None
    docs: Optional[AgentDocs] = None
    webhooks: Optional[list[AgentWebhook]] = None
    metadata: Optional[dict[str, str]] = None
    signature: Optional[str] = None


@dataclass
class AgentInternalAPI:
    docs_url: Optional[str] = None
    endpoints: Optional[list[AgentInterface]] = None
    details: Optional[dict[str, str]] = None


@dataclass
class ExtendedAgentCard:
    agent_id: str
    agent: AgentCard
    updated_at: str
    private_skills: Optional[list[str]] = None
    rate_limits: Optional[dict[str, str]] = None
    internal_api: Optional[AgentInternalAPI] = None
    metadata: Optional[dict[str, str]] = None


@dataclass
class AgentQueryParams:
    q: Optional[str] = None
    skill: Optional[str] = None
    capability: Optional[str] = None
    tag: Optional[str] = None
    tags: Optional[list[str]] = None
    username: Optional[str] = None
    crypto_id: Optional[str] = None
    network: Optional[str] = None
    asset: Optional[str] = None
    max_amount: Optional[str] = None
    group: Optional[str] = None
    limit: Optional[int] = None
    offset: Optional[int] = None


@dataclass
class ResolveResponse:
    identity: Optional[Identity]
    agent: Optional[AgentCard] = None


@dataclass
class ReverseResponse:
    crypto_id: str
    identities: list[Identity]
    agents: Optional[list[AgentCard]] = None

File: sdk/src/types/directory.ts
import type { Identity, PaymentMethod } from "./identity.js";

export interface AgentInterface {
  url: string;
  binding: string;
  version: string;
}

export interface AgentPayment {
  network: string;
  asset: string;
  rateType: string;
  amount: string;
}

export interface AgentDocs {
  swaggerJson?: string;
  swaggerMd?: string;
  skillMd?: string;
  swaggerJsonUrl?: string;
  swaggerMdUrl?: string;
  skillMdUrl?: string;
}

export interface AgentWebhook {
  event: string;
  url: string;
  secretRef?: string;
  description?: string;
  metadata?: Record<string, string>;
}

export interface AgentCard {
  agentId: string;
  name: string;
  description?: string;
  username?: string;
  cryptoId: string;
  publicKey?: string;
  url?: string;
  endpoint?: string;
  supportedInterfaces?: Array<AgentInterface>;
  skills?: Array<string>;
  capabilities?: Array<string>;
  tags?: Array<string>;
  paymentMethods?: Array<PaymentMethod>;
  paymentRequirements?: AgentPayment;
  groups?: Array<string>;
  docs?: AgentDocs;
  webhooks?: Array<AgentWebhook>;
  metadata?: Record<string, string>;
  signature?: string;
  createdAt: string;
  updatedAt: string;
}

export interface AgentInternalAPI {
  docsUrl?: string;
  endpoints?: Array<AgentInterface>;
  details?: Record<string, string>;
}

export interface ExtendedAgentCard {
  agentId: string;
  agent: AgentCard;
  privateSkills?: Array<string>;
  rateLimits?: Record<string, string>;
  internalApi?: AgentInternalAPI;
  metadata?: Record<string, string>;
  updatedAt: string;
}

export interface AgentQueryParams {
  q?: string;
  skill?: string;
  capability?: string;
  tag?: string;
  tags?: Array<string>;
  username?: string;
  cryptoId?: string;
  network?: string;
  asset?: string;
  maxAmount?: string;
  group?: string;
  limit?: number;
  offset?: number;
}

export interface ResolveResponse {
  identity: Identity | null;
  agent?: AgentCard;
}

export interface ReverseResponse {
  cryptoId: string;
  identities: Array<Identity>;
  agents?: Array<AgentCard>;
}
